### ISSUE TITLE
poly lines vs paths in SVG

### ISSUE THREAD

**Issue by sirogi** [appmancer/JXPlotterXVG#1]
Hello! 

Been testing you're excellent translator, stumbled on a couple of things.
Paths drawn in Inkscape (SVG path entity) generate a slightly different Gcode than polylines.
For polylines,  I seem to be missing the pen up / down  (laser on / off) codes in the right places.

(let me know if you need the sample files, svg and generated gcode)

Also, I  still had to find / replace the laser on / off codes to pen up / down. (M300 S160 and M300 S100.00) that I ripped off the pinecone own Xplotter.g file.  if you're not handling a switch of tools yet, maybe a command line option could be handy, format the gcode for a brush or laser.

Thank you for a very useful piece of code.

**Comment by sirogi** [comment]
Here are the source SVG vs generated Gcode ... of version 1.0.0 ;)



[Pline_vs_path_tests.zip](https://github.com/appmancer/JXPlotterXVG/files/1486063/Pline_vs_path_tests.zip)

**Comment by appmancer** [comment]
[charles.zip](https://github.com/appmancer/JXPlotterXVG/files/1487134/charles.zip)
Morning, there was a bug where I've written the method to set the head up and down on polylines, but it was empty!  I've fixed it.
The tool should automatically be set by the material.  1 is pen (or the words 'pen' or 'brush') and 2 is the laser (or the word 'laser'). 
I've included the latest binary (now in GitHub as well), and 2 material files and their output.  Hopefully that will be all correct now!

**Comment by sirogi** [comment]
Great! Can't wait to try it.

Cheers.

-c

**Comment by sirogi** [comment]
Had a Look, certainly can leave my search and replace behind, thanks! 
Interestingly you're using M300 S90 and S170 for pen up/down
I ripped off the sample Gcode that pinecone Used M300 S100 and S160
wonder is the numbers map to pen depth or something.

I did notice with your current version, at the end of a trace, the poly line does do a pen up before returning to origin, while the path doesn't (yikes! :) imagine a nice long diagonal after hours of.. well..)

One thing I wanted to share, while playing this weekend, didn't want to wait for the slower G0 FR1500 commands, so ran everything at FR 3500, The plotter was in such a hurry rushing from one location to another, it did pen up in motion. You get sharp corners at end of traces. cool.. but useless..
So I inserted 20ms pauses after pen up. Solved it. Might have accelerated everything a little bit even. 

Thanks for the amazing response. I can ditch my vim search and replace routine, one less step.

Thanks again!
